### PR TITLE
Fix generate_key

### PR DESCRIPTION
**Pull request by NathanReb**
We should add basic tests that just performs a full key exchange to catch that kind of stupid/horrible mistake.

### CODE REVIEW

**Comment by NathanReb** [comment]
I just added such a test in a separate commit. Let me know what you think and let's merge and cut a release, I feel dirty and ashamed that I released such a monster!

File: test/test_fiat_p256.ml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    else
+      go ((f i)::acc) (i + 1)
+  in
+  go [] 0
+
+let int32_to_hex i = Printf.sprintf "%08Lx" i
+
+let prng len =
+  let i32_count = len / 4 + 1 in
+  let i32s = list_init i32_count (fun _ -> Random.int64 0x100000000L) in
```
Comment (emillon) [inline test/test_fiat_p256.ml]: I think you'll want a toplevel `let () = Random.self_init ()` somewhere (that does not matter too much).

**Comment by NathanReb** [comment]
I added the `Random.self_init ()`!

**Comment by emillon** [comment]
Thanks!

**Pull request merged**